distsql/request_builder.go

```diff
@@ -58,12 +58,10 @@ func (builder *RequestBuilder) Build() (*kv.Request, error) {
 			},
 		}
 	}
-	failpoint.Inject("assertRequestBuilderStalenessOption", func(val failpoint.Value) {
+	failpoint.Inject("assertRequestBuilderReplicaOption", func(val failpoint.Value) {
 		assertScope := val.(string)
-		if len(assertScope) > 0 {
-			if builder.IsStaleness && assertScope != builder.ReadReplicaScope {
-				panic("request builder get staleness option fail")
-			}
```

Comment:
The reason why we didn't find bug by unit test for follower read is due to the asserting condition is made for stale read, thus we change the asserting condition in order to detect follower read's situation.